### PR TITLE
CMR-7362 - Adding scrolling example

### PR DESCRIPTION
**Pull request by jceaser**
Adding a script which demonstrates how to do a scrolling query against CMR.
Other changes:
* adding logging used while developing and a function to ensure that tokens are not printed in clear text
* other pylint (using with) changes
* Found misleading info in the token example notebook

The script can be run with: 

    >python3 ./demos/scripts/scroll.py         
    returned items: 214
    uniq keys: 214

### CODE REVIEW

**Review by mmorahan — APPROVED** [review]
Tested the scroll example script with additional fields for my own work...much faster than paging.

**Comment by sritz-23** [comment]
The script works for me. However, I get some sort of git error. Still works. 

`saritz@mbp-saritz ~ % /usr/local/bin/python3 /Users/saritz/Desktop/test_scroll.py                   fatal: Not a git repository (or any of the parent directories): .git
Could not create build info: Command '['git', 'rev-parse', '--short', 'HEAD']' returned non-zero exit status 128.
returned items: 4066
uniq keys: 4066`

**Comment by jceaser** [comment]
scott, your error is because the script is running outside of the git repository. It can be recreated, but is not related to this change. This behavior is pre-existing and we should write a new ticket for researching how to make this warning go away.

**Comment by mmorahan** [comment]
Thomas, my script is running outside my git hub directory and I don't get Scott's git error.

File: CMR/python/cmr/util/common.py [hunk 1]
```diff
@@ -144,3 +144,19 @@ def help_format_lambda(contains=""):
     # n=name, c=content ; made short to keep line length down and pylint happy
     out = lambda n, c : (layout.format(n, c.__doc__.strip())) if contains in n else ""
     return out
+
+def mask_dictionary(data, keys):
+    """
+    Prevent sensitive information from being printed by masking part of a
+    dictionaries content
+    Return a copy of the
```
Comment (eereiter) [inline CMR/python/cmr/util/common.py:152]: Please finish the sentence.

**Review by eereiter — APPROVED** [review]

**Pull request merged**